invenio_app_ils/closures/api.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""ILS location closures API."""
+
+from datetime import timedelta
+
+import arrow
+
+from invenio_app_ils.errors import IlsException
+from invenio_app_ils.proxies import current_app_ils
+
+_ONE_DAY_INCREMENT = timedelta(days=1)  # Atomic increment
+
+
+def _is_normally_open(location, date):
+    """Checks if the location is normally opened on a given date wrt to the regular schedule."""
+    opening = location["opening_weekdays"]
+    opening_weekday = opening[date.weekday()]
+    return opening_weekday["is_open"]
+
+
+def _is_exceptionally_open(location, date):
+    """Checks if the location is exceptionally opened on a given date, or None if undefined."""
+    for exception in location["opening_exceptions"]:
+        if date <= arrow.get(exception["end_date"]).date():
+            if arrow.get(exception["start_date"]).date() <= date:
+                return exception["is_open"]
+            else:
+                return None  # Early return
```

Comment:
I am not sure about this early return... isn't it working only if exceptions are sorted by `end_date` `ASC`?

Otherwise, I would do:
What about:
```
for exception in location["opening_exceptions"]:
    start_date = arrow.get(exception["start_date"]).date()
    end_date = arrow.get(exception["end_date"]).date()
    date_in_range = start_date <= date <= end_date
    if date_in_range:
        return exception["is_open"]
return False
```
I am also not sure why returning `None` instead of `False`

Reply:
The function can return three different results:
- `True | False`: an interval was found and its value is returned
- `None`: no interval was found

It's important to make the distinction between `False` and `None` because they should not be treated in the same way. More precisely if no interval matches, we should branch to the weekdays procedure.

Reply:
I find it  a bit dangerous, because we need to remember about it in all the conditional statements, which are to come based on this function. The question is if it is exceptionally open, intuition says the answer should be yes/no. Is the fact of not finding the exception changing anything? We still know that it is not exceptionally open.

Reply:
Because it can be either exceptionally open or exceptionally closed, in which case we would completely ignore the weekly schedule.

Reply:
I still do not see the need for `None`. I think:
```
is_normally_open XOR is_exceptionally_closed XOR is_exceptionally_open 
```
covers all the cases, no?